index.js

```diff
@@ -207,6 +207,32 @@ HttpPushRgb.prototype = {
             .setCharacteristic(Characteristic.FirmwareRevision, FIRMWARE_REVISION);
 
         switch (this.serviceCategory) {
+            case 'Fan':
+                this.log('Creating Fan');
+                this.service = new Service.Fan(this.name);
+
+                if (this.switch.status) {
+                    this.service
+                        .getCharacteristic(Characteristic.On)
+                        .on('get', this.getPowerState.bind(this))
+                        .on('set', this.setPowerState.bind(this));
+                } else {
+                    this.service
+                        .getCharacteristic(Characteristic.On)
+                        .on('set', this.setPowerState.bind(this));
+                }
+
+                // Handle fan speed
+                if (this.has.brightness) {
```

Comment:
Why not create a new attribute fan_speed?

Reply:
I could use most of the existing code from brightness and create a new attribute for fan_speed. It may be the proper way regarding the readabilty of the codebase but is it really necessary? The values we get from brightness are analogous to the values we need for fan speed.